src/components/MultiHopTrade/components/VerifyAddresses/VerifyAddresses.tsx

```diff
@@ -219,8 +220,23 @@ export const VerifyAddresses = () => {
     ],
   )
 
-  const handleBuyVerify = useCallback(() => handleVerify('buy'), [handleVerify])
-  const handleSellVerify = useCallback(() => handleVerify('sell'), [handleVerify])
+  const checkLedgerAppOpen = useLedgerOpenApp()
+
+  const handleBuyVerify = useCallback(async () => {
+    // Only proceed to verify the buy address if the promise is resolved, i.e the user has opened
+    // the Ledger app without cancelling
+    await checkLedgerAppOpen(buyAsset.chainId)
+      .then(() => handleVerify('buy'))
+      .catch(console.error)
+  }, [checkLedgerAppOpen, handleVerify, buyAsset.chainId])
+
+  const handleSellVerify = useCallback(async () => {
+    // Only proceed to verify the sell address if the promise is resolved, i.e the user has opened
+    // the Ledger app without cancelling
+    await checkLedgerAppOpen(sellAsset.chainId)
+      .then(() => handleVerify('sell'))
+      .catch(console.error)
```

Comment:
preferably-blocking: I think we should do this under the hood instead of repeating it all over the app

Reply:
We have a wallet in the context of `handleVerify`, so we could indeed do the await in there instead so we don't have this pattern everywhere.

Reply:
I did try to improve this, but we actually need allow the parent to handle any errors, I doubt we can do super better with this